### ISSUE TITLE
Validating the model

### ISSUE THREAD

**Issue by NiAmL** [ThilinaRajapakse/pytorch-transformers-classification#20]
Hi,

I would like to know if the model over-fits and also the optimum number of epochs,  plotting accuracy and loss as it's shown [here](https://stackoverflow.com/questions/41908379/keras-plot-training-validation-and-test-set-accuracy). It would be possible to do it using this repo without making too many changes (maybe using the evaluation results as validation)?

Thanks.

**Comment by ThilinaRajapakse** [comment]
You can get the training loss without any changes. You can use `tensorboardx` to get a graph of the training loss. The loss information is being written to the 'runs' directory.

If you want to evaluate on the dev set during training, you can set `evaluate_during_training` to True in the args dict.

If you want to add additional information to that, you can use additional `tb_writer.add_scalar()` calls inside the train function.

**Comment by NiAmL** [comment]
The concept is simple but I am still not able to plot anything. 

In the training function we have these lines:

`tb_writer.add_scalar('eval_{}'.format(key), value, global_step)`
`tb_writer.add_scalar('lr', scheduler.get_lr()[0], global_step)`
`tb_writer.add_scalar('loss', (tr_loss - logging_loss)/args['logging_steps'], global_step)`

If I understand it right, for each epoch, the first one contains every result that we have generated, and the second and third ones the learning rate and loss. 

For this case I am going to plot this info only with one epoch, but it should still show something right? As per the documentation, I understand that we only need to launch this line `tensorboard --logdir runs` (as we store the scalars in the `runs` directory), am I right?

I get no error message in any point of the implementation (having activated the option `evaluate_during_training`), but when I try to plot it I get this error:

<img width="960" alt="Untitled" src="https://user-images.githubusercontent.com/50815888/66448154-aa8e3e00-ea76-11e9-8635-25cbb9d5b8fa.png">

There is a folder called `runs` in the experiment folder.

**Comment by ThilinaRajapakse** [comment]
There should be a subdirectory inside runs for every training run. So your command would look like `tensorboard --logdir=runs/subdirectory`.

To visualize the last run, you can use the line below.
`tensorboard --logdir=$(ls -td | head -1)`

**Comment by NiAmL** [comment]
There is a directory called `Oct09_03-14-56_31dd366812b4`, but when I run this line:

`!tensorboard --logdir="runs/Oct09_03-14-56_31dd366812b4" --host localhost --port 8088`

I get a message saying that ` site http://localhost:8088/ can't be reached, localhost refused to connect.`

I have tried with different ports but no way. I have been researching on the internet and some people says that it's possible to achieve this using a tunnel, ngrok, [here](https://www.dlology.com/blog/quick-guide-to-run-tensorboard-in-google-colab/). Before trying it I would like to ask you if it makes sense, or if it should work straightforward from google colab.

**Comment by NiAmL** [comment]
Supposedly wouldn't be needed... 

https://www.tensorflow.org/tensorboard/tensorboard_in_notebooks

**Comment by NiAmL** [comment]
ok, I think I got it.

First I loaded the TensorBoard notebook extension
`%load_ext tensorboard`

The I set to use the version 2:
`%tensorflow_version 2.x`

And run the tensorboard with:
`%tensorboard --logdir=runs/Oct09_03-14-56_31dd366812b4`

So I got the dashboard. 

<img width="958" alt="Untitled" src="https://user-images.githubusercontent.com/50815888/66453888-f3042680-ea8b-11e9-815a-72deb310d399.png">

I need to play with it a bit more to see if it's working, but looks it is.

**Comment by ThilinaRajapakse** [comment]
Great to see you got it to work. I didn't realize you were on Colab!
